flytekit/interaction/click_types.py

```diff
@@ -305,10 +305,12 @@ def __init__(self, python_type: typing.Type):
         self._python_type = python_type
 
     def _parse(self, value: typing.Any, param: typing.Optional[click.Parameter]):
-        if type(value) == dict or type(value) == list:
+        if isinstance(value, dict) or isinstance(value, list):
             return value
         try:
-            return json.loads(value)
+            if isinstance(value, str):
+                return json.loads(value)
+            return json.loads(json.dumps(value, default=lambda o: o.__dict__))
```

Comment:
what's the `type` of `value` will be here?
is it only dataclass?
if only dataclass, can we use 
`import dataclasses; dataclasses.is_dataclass(something)` to check the value first?

https://stackoverflow.com/questions/56106116/check-if-a-class-is-a-dataclass-in-python

Reply:
Based on the test results, the type of the value corresponds to the user-defined class.
For instance, in the first example above, the type of the value will be `Foo`.